BSP/STM32F413/Src/BSP_CAN.c

```diff
@@ -2,10 +2,13 @@
 #include "stm32f4xx.h"
 
 #define CAN_MODE        CAN_Mode_Normal
+#define CANFIFOMAX  32 //most messages it can take before wrapping around
 
 static CanTxMsg TxMessage;
 static CanRxMsg RxMessage;
-
+uint8_t CANFIFO[32][8] = {0}; //initialize 32 by 8 array containing FIFO data 
+CANFIFOHEAD = 0; //row of head
+CANFIFOTAIL = 0; //row of tail
```

Comment:
When declaring a variable, make sure to specify the type. In this case, `static int`.

Also, since these are not `const`, they should be named something like `rxFifoHead` and `rxFifoTail`.